File: 3531-count-covered-buildings/3531-count-covered-buildings.cpp
using ll = long long;
class Solution {
public:
    ll getid(vector<int>& x) {
        return (ll)x[0] * 100000 + x[1];
    }
    int countCoveredBuildings(int n, vector<vector<int>>& buildings) {
        int m = buildings.size();
        unordered_map<ll, bool> ok;
        vector<vector<int>> xAxis = buildings; 
        vector<vector<int>> yAxis = buildings; 
               
        for(auto building : buildings) {
            ok[getid(building)] = true;
        }

        sort(xAxis.begin(), xAxis.end(), [](const auto &a, const auto &b){
            if(a[0] == b[0]) {
                return a[1] < b[1];
            }    
            return a[0] < b[0];
        });
        sort(yAxis.begin(), yAxis.end(), [](const auto &a, const auto &b){
            if(a[1] == b[1]) {
                return a[0] < b[0];
            }    
            return a[1] < b[1];
        });

        for(int i = 0; i < m; i++) {
            if(i == 0 || i == m-1 || xAxis[i-1][0] != xAxis[i][0] || xAxis[i+1][0] != xAxis[i][0]) ok[getid(xAxis[i])] = false;
            if(i == 0 || i == m-1 || yAxis[i-1][1] != yAxis[i][1] || yAxis[i+1][1] != yAxis[i][1]) ok[getid(yAxis[i])] = false;
        }

        int ans = 0;
        for(const auto &[_, v] : ok) {
            ans += v;
        }
        return ans;
    }
};

File: 3531-count-covered-buildings/3531-count-covered-buildings.go
func countCoveredBuildings(n int, buildings [][]int) int {
    maxX := make([]int, n+1)
    minX := make([]int, n+1)
    maxY := make([]int, n+1)
    minY := make([]int, n+1)

    for i := 0; i < n+1; i++ {
        minX[i] = math.MaxInt
        minY[i] = math.MaxInt
    }

    for _, b := range buildings {
        x, y := b[0], b[1]
        maxX[x] = max(maxX[x], y)
        minX[x] = min(minX[x], y)
        maxY[y] = max(maxY[y], x)
        minY[y] = min(minY[y], x)
    }

    ans := 0
    for _, b := range buildings {
        x, y := b[0], b[1]
        if y > minX[x] && y < maxX[x] && x > minY[y] && x < maxY[y] {
            ans++
        }
    }

    return ans
}